ttnn/cpp/ttnn/operations/pool/generic/generic_pools.cpp

```diff
@@ -2,28 +2,40 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include "max_pool2d.hpp"
+#include "generic_pools.hpp"
 
 #include "impl/buffers/buffer_constants.hpp"
-#include "ttnn/operations/conv/conv2d/conv2d.hpp"
-#include "ttnn/operations/sliding_window/sliding_window.hpp"
 #include "tt_metal/common/math.hpp"
 #include "ttnn/common/constants.hpp"
+#include "ttnn/operations/conv/conv2d/conv2d.hpp"
+#include "ttnn/operations/sliding_window/sliding_window.hpp"
 
 namespace ttnn {
 namespace operations::pool {
 
-Tensor MaxPool2DOp::invoke(uint8_t queue_id,
-                           const Tensor& input_tensor,
-                           uint32_t batch_size,
-                           uint32_t input_h, uint32_t input_w,
-                           uint32_t channels,
-                           std::array<uint32_t, 2> kernel_size,
-                           std::array<uint32_t, 2> stride,
-                           std::array<uint32_t, 2> padding,
-                           std::array<uint32_t, 2> dilation,
-                           const std::optional<const MemoryConfig> memory_config,
-                           const std::optional<const TensorMemoryLayout> applied_shard_scheme) {
+namespace {
+
+float get_pool_pad_value(Pool2DType pool_type) {
+    switch (pool_type) {
+        case Pool2DType::MAX_POOL2D: return -std::numeric_limits<float>::infinity();
```

Comment:
`0xf7ff` is the bfloat16 -inf value, so I don't think the above float -inf will work.

Reply:
I reworked them to use `bfloat16(value).to_packed()` to properly convert the float32 into bf16